src/JMS/Serializer/Construction/DoctrineObjectConstructor.php

```diff
@@ -87,11 +88,19 @@ public function construct(VisitorInterface $visitor, ClassMetadata $metadata, $d
         $identifierList = array();
 
         foreach ($classMetadata->getIdentifierFieldNames() as $name) {
-            if (!array_key_exists($name, $data)) {
+            if (method_exists($visitor, 'getNamingStrategy')) {
```

Comment:
cant you check if `$visitor instanceof AbstractVisitor`?

Reply:
i`m going to change this. but i think it will break the current master since the method "getNamingStrategy" was removed.
See: https://github.com/schmittjoh/serializer/commit/9e3fd4d83a0e81d97da2f2cb12568a9858952007#diff-04059dbc4f97e37925ef18e828ce4f5b

i`m going to rework this merge request this Sunday.
i hope this pull request helps you to pinpoints the error from #806